### PR TITLE
[Update Detekt Version] Update Detekt to 1.16.0

### LINKED ISSUE #16894: Update Detekt Version - Update to 1.16.0
Parent #16892

This issue is about updating Detekt to the next version, the [1.16.0](https://github.com/detekt/detekt/releases/tag/v1.16.0).

While updating Detekt to the next version, new rules might end up being enabled that would need to be resolved/suppressed as well. This issue will deal with that too.

### PR DESCRIPTION
**Pull request by ParaskP7**
Parent: #16892
Closes: #16894

This PR:
- Updates Detekt to `1.16.0` (from `1.15.0`) (3efb9e86bf0c69bdf03560a06e0df37432266a5e).
- Resolves/Suppresses new warnings that started appearing due to the fact that the below rules are now enabled by default (previously disabled by default):
    - `PrintStackTrace` (exceptions)
    - `SwallowedException` (exceptions)
    - `ImplicitDefaultLocale` (potential-bugs)
    - `SerialVersionUIDInSerializableClass` (style)
    - `UnusedPrivateMember` (style)
    - `VarCouldBeVal` (style)

PS: I recommend reviewing this PR commit-by-commit.

-----

To test:

- Verify that all the CI checks are successful (especially the `detekt` check).
- Smoke test the app and verify everything works as expected.

-----

## Regression Notes
1. Potential unintended areas of impact

N/A

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt [hunk 1]
```diff
@@ -374,7 +374,7 @@ class RestoreViewModel @Inject constructor(
         val currentIndex = wizardManager.currentStep
         val targetIndex = wizardManager.stepPosition(targetStep)
 
-        for (i in currentIndex downTo targetIndex) {
+        for (it in currentIndex downTo targetIndex) {
```
Comment (oguzkocer) [inline WordPress/src/main/java/org/wordpress/android/ui/jetpack/restore/RestoreViewModel.kt]: Using `it` makes me uneasy. Do you think the following would be better, so we can use `_` instead?

```suggestion
        (currentIndex downTo targetIndex).forEach { _ ->
```

File: libs/image-editor/src/main/kotlin/org/wordpress/android/imageeditor/crop/CropFragment.kt [hunk 2]
```diff
@@ -153,7 +153,7 @@ class CropFragment : Fragment(), UCropFragmentCallback {
         }
     }
 
-    private fun showCropError(errorMsg: String?, errorResId: Int) {
+    private fun showCropError(errorResId: Int) {
         // TODO: track exact error errorMsg
```
Comment (oguzkocer) [inline libs/image-editor/src/main/kotlin/org/wordpress/android/imageeditor/crop/CropFragment.kt]: I think we should remove the `TODO` if we are removing the `errorMsg` parameter.

```suggestion
```

**Review by oguzkocer — APPROVED** [review]
@ParaskP7 Thank you for the changes. I left a couple suggestions, but in general the changes look good to me.

One small bit of feedback; I thought the commit list was quite noisy with a lot of commits with just a single line change for a `@SuppressWarnings("unused")` or similar without any extra context in the commit message. I think it would have been better to combine at least some of these commits together as I don't see any practical benefit to separating them. An option was to combine all the `@Suppress` changes together in one commit and add a line comment to the PR where any extra context is necessary. This would have made reviewing the PR easier and keep our history cleaner.

**Comment by ParaskP7** [comment]
> @ParaskP7 Thank you for the changes. I left a couple suggestions, but in general the changes look good to me.

👋 @oguzkocer and thank you so much for reviewing and testing this PR! 🙇 🥇 

> One small bit of feedback; I thought the commit list was quite noisy with a lot of commits with just a single line change for a @SuppressWarnings("unused") or similar without any extra context in the commit message. I think it would have been better to combine at least some of these commits together as I don't see any practical benefit to separating them. 

Thank you so much for this feedback Oguz, you are totally right, I could have combined all `@SuppressWarnings("unused")` or `@Suppress("unused")` changes in one commit. It is just that they way I was dealing with resolving/suppressing all those `UnusedPrivateMember` warnings (31 of them) was one-by-one as for some I was able to resolve them, but for others I could only suppress them (for example, event bus, kotlin keyword, todo related, etc).

Since I am now seeing this made it difficult for you to review, I understand that I could have stop from committing the suppress related changes and gather those in one last commit at the end, while at the same time committing the resolved related changes.

Please expect me to do so next time and thank YOU! 💯 

> An option was to combine all the @Suppress changes together in one commit and add a line comment to the PR where any extra context is necessary. This would have made reviewing the PR easier and keep our history cleaner.

👍

**Pull request merged**